pkg/ast/spl/spl.peg

```diff
@@ -961,12 +961,19 @@ TcOptionCMD <- option:("usenull" / "useother" / "nullstr" / "otherstr") {
     return optionStr, nil
 }
 
-BinSpanLenOption <- number:(FloatAsString / IntegerAsString) timeScale:(TimeScale)? {
+BinTimeScale <- timeUnit:(Second / Month / Subseconds / Minute / Hour / Day / Week / Quarter / Year) {
```

Comment:
Did you refactored the `TimeScale` to `BinTimeScale`?

Reply:
TimeScale does not have Year in it. So had to create a similar one. I also cannot use the one from RelativeTimestamp because it does not have subseconds.

Reply:
got it. But I think you can still use the same one like this:

```
BinTimeScale <- timeUnit: (TimeScale / Year) 
```

And for RelTimeUnit

```
RelTimeUnit <- timeUnit: (TimeScale) !(Subseconds)
```

Or even better we can refactor this TimeUnit like this:

```
AllTimeScale <- timeUnit:(Second / Month / Subseconds / Minute / Hour / Day / Week / Quarter / Year) {
    return timeUnit, nil
}

TimeScale <- timeUnit:(AllTimeScale) !(Year) {
    return timeUnit, nil
}

BinTimeScale <- timeUnit:(AllTimeScale) {
    return timeUnit, nil
}

RelTimeUnit <- timeUnit:(AllTimeScale) !(Subseconds) {
    return strconv.Itoa(int(timeUnit.(utils.TimeUnit))), nil
}

```

So in this way `AllTimeScale` contains all the time unit and the other will specify what they do not want to match.

Reply:
I think that's a good idea, but I'm not sure this will do what you want:
```
RelTimeUnit <- timeUnit: (TimeScale) !(Subseconds)
```
I _think_ that rule would match any `TimeScale` (including `Subseconds`) followed by anything that's not `Subseconds`—but I might be wrong

If that doesn't work, another option would be to use `AllTimeScale` everywhere and then inside the Go part of the rule check to ensure the `AllTimeScale` value is valid for the particular command you're implementing

Reply:
addressed in the new commit.

Reply:
Did you change `TimeScale` also? I don't see it on the changes?

Reply:
Yes I have removed the TimeScale as it was used only at one place. I have added the check to throw error if timeUnit is year in the SpanLength rule that was using it. I have also added a parser test which verifies this.